repose-aggregator/components/filters/body-patcher-filter/src/main/scala/org/openrepose/filters/bodypatcher/BodyPatcherFilter.scala

```diff
@@ -0,0 +1,150 @@
+          new HttpServletRequestWrapper(httpRequest, new ServletInputStreamWrapper(new ByteArrayInputStream(PJson.stringify(patchedValue).getBytes)))
+        } else {
+          httpRequest
+        }
+      case Xml =>
+        //todo: implement xml support
+        httpRequest
+      case Other =>
+        httpRequest
+    }
+
+
+    //check if we might do work to the response
+    if (responsePatches.isEmpty) {
+      chain.doFilter(patchedRequest, httpResponse)
+    } else {
+      val wrappedResponse: HttpServletResponseWrapper = new HttpServletResponseWrapper(httpResponse, PASSTHROUGH, MUTABLE)
+      chain.doFilter(patchedRequest, wrappedResponse)
+      wrappedResponse.uncommit()
+
+      determineContentType(wrappedResponse.getContentType) match {
+        case Json =>
+          val jsonPatches: List[String] = filterJsonPatches(responsePatches)
+          if (jsonPatches.nonEmpty) {
+            logger.debug("Applying json patches on the response")
+
+            val contentStream = wrappedResponse.getOutputStreamAsInputStream
+            val contentValue = if (contentStream.available() > 0) {
+              PJson.parse(contentStream)
+            } else {
+              PJson.obj()
+            }
+            val patchedValue: JsValue = applyJsonPatches(contentValue, jsonPatches)
+            wrappedResponse.setOutput(new ByteArrayInputStream(PJson.stringify(patchedValue).getBytes))
+          }
+        case Xml => //todo implement xml support
+        case Other =>
+      }
+      wrappedResponse.commitToResponse()
+    }
+  }
+
+  def filterChanges(request: HttpServletRequest): List[ChangeDetails] = {
+    val urlPath: String = new URL(request.getRequestURL.toString).getPath
+    configuration.getChange.asScala.toList
+        .filter(_.getPath.r.findFirstIn(urlPath).isDefined)
+  }
+
+  def filterRequestChanges(changes: List[ChangeDetails]): List[Patch] = {
+    changes.flatMap(change => Option(change.getRequest))
+  }
+
+  def filterResponseChanges(changes: List[ChangeDetails]): List[Patch] = {
+    changes.flatMap(change => Option(change.getResponse))
+  }
+
+  def determineContentType(contentType: String): ContentType = {
+    //magic code from stack overflow, i'm a terrible person, http://stackoverflow.com/questions/4636610/how-to-pattern-match-using-regular-expression-in-scala
+    implicit class Regex(sc: StringContext) {
+      def r = new util.matching.Regex(sc.parts.mkString, sc.parts.tail.map(_ => "x"): _*)
+    }
+
+    Option(contentType).getOrElse("").toLowerCase match {
+      case r".*json.*" => Json
+      case r".*xml.*" => Xml
+      case _ => Other
+    }
+  }
+
+  def filterJsonPatches(patches: List[Patch]): List[String] = {
+    patches.flatMap(patch => Option(patch.getJson))
+  }
+
+  def applyJsonPatches(body: JsValue, patches: List[String]): JsValue = {
+    patches.foldLeft(body)((content: JsValue, patch: String) => {
+      val jsPatch: JsonPatch = JsonPatch.parse(patch)
+      jsPatch(content)
+    })
```

Comment:
Yeah, applying them all makes composition way easier. otherwise you have to get very explicit and repeat yourself a bunch potentially.